Move old fastpath into a conditional that's enabled via an attribute, go to sdpa path by default

mikekgfb opened the pull request:
Summary: Move old fastpath into a conditional that's enabled via an attribute, go to sdpa path by default

Test Plan: sandcastle

Differential Revision: D41590030

cpuhrsch reviewed (changes requested):
I think we can resolve that by making the default path match performance to the fast path. We're close to that.

Commit:
Move old fastpath into a conditional that's enabled via an attribute, go to sdpa path by default (#89894)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/89894

Move old fastpath into a conditional that's enabled via an attribute, go to sdpa path by default

Test Plan: sandcastle

Differential Revision: D41590030

fbshipit-source-id: 4f815a3cf9967cc125bcd9f8a36734507069063b
